state_machine/applications/system-check/tests/imu_test.py

```diff
@@ -0,0 +1,147 @@
+
+        acc = pycubed.acceleration
+        gyro = pycubed.gyro
+        starting_mag = pycubed.magnetic
+        time.sleep(mag_time)
+        ending_mag = pycubed.magnetic
+        # calculate the change in the total magnetometer reading
+        mag = (sqrt(ending_mag[0] ** 2 + ending_mag[1] ** 2 + ending_mag[2] ** 2) -
+               sqrt(starting_mag[0] ** 2 + starting_mag[1] ** 2 + starting_mag[2] ** 2))
+        print("Data Collection Complete")
+        return acc, gyro, mag
+
+
+def imu_test(cubesat, result_dict, testidx):
+    """
+    All automation happens in this function
+    Select burnwire and voltage level
+    Process user test results and update the result dictionary accordingly
+    """
+    if testidx == "Still":
+        # record acceleration
+        acc, gyro, mag = user_test(cubesat, result_dict, testidx)
+        result_val_string = "Testing IMU when cubesat is still. Acc: " + str(acc)
+
+        # if total acceleration is within 0.2 m/s^2 of 9.8, true
+        if abs(9.8 - sqrt(acc[0] ** 2 + acc[1] ** 2 + acc[2] ** 2)) < 0.2:
+            result_dict['IMU_Acc_Still'] = (result_val_string, True)
+        else:  # else false
+            result_dict['IMU_Acc_Still'] = (result_val_string, False)
+
+    elif testidx == "Moving":
+        # record acceleration
+        acc, gyro, mag = user_test(cubesat, result_dict, testidx)
+        result_val_string = "Testing IMU when cubesat is being moved. Acc: " + str(acc)
+
+        # if each acceleration component is more than 0, true
+        if abs(acc[0]) > 0 and abs(acc[1]) > 0 and abs(acc[2]) > 0:
+            result_dict['IMU_Acc_Moving'] = (result_val_string, True)
+        else:  # else false
+            result_dict['IMU_Acc_Moving'] = (result_val_string, False)
+
+    elif testidx == "Rotating":
+        # record gyro reading
+        acc, gyro, mag = user_test(cubesat, result_dict, testidx)
+        result_val_string = "Testing IMU when cubesat is rotating on a turntable. Gyro: " + str(gyro)
+
+        # if total gyro reading is within 2 deg/s^2 of angular velocity, return true
+        ang_vel = float(input("Please enter the speed the turntable was rotating at in deg/s: "))
+        if abs(ang_vel - sqrt(gyro[0] ** 2 + gyro[1] ** 2 + gyro[2] ** 2)) < 2:
+            result_dict['IMU_Gyro_Turntable'] = (result_val_string, True)
+        else:  # else false
+            result_dict['IMU_Gyro_Turntable'] = (result_val_string, False)
+
+    elif testidx == "Magnet":
+        # record magnetometer reading
+        acc, gyro, mag = user_test(cubesat, result_dict, testidx)
+        result_val_string = ("Testing IMU when cubesat is subject to an external magnetic field." +
+                             "Final Mag Reading: " + str(mag))
+
+        # if change in magnetometer reading is more than 0, true
+        if mag > 0:
+            result_dict['IMU_Gyro_Turntable'] = (result_val_string, True)
```

Comment:
Copy-paste error here, this shouldn't be `IMU_Gyro_Turntable`

Reply:
fixed